lib/matplotlib/backends/backend_qt4agg.py

```diff
@@ -72,16 +45,6 @@ class FigureCanvasQTAgg(FigureCanvasQTAggBase,
 
     """
 
-    def __init__(self, figure):
-        if DEBUG:
-            print('FigureCanvasQtAgg: ', figure)
-        FigureCanvasQT.__init__(self, figure)
```

Comment:
This is further handled by the qtcairo commit, but in fact with the new MRO (specifically, due to FigureCanvasQT going after FigureCanvasAgg), this is not needed anymore, because the MRO FigureCanvasQT(4)Agg is
```
matplotlib.backends.backend_qt4agg.FigureCanvasQTAgg
matplotlib.backends.backend_qt5agg.FigureCanvasQTAggBase
matplotlib.backends.backend_agg.FigureCanvasAgg
matplotlib.backends.backend_qt4.FigureCanvasQT
matplotlib.backends.backend_qt5.FigureCanvasQT
PyQt4.QtGui.QWidget
PyQt4.QtCore.QObject
sip.wrapper
PyQt4.QtGui.QPaintDevice
sip.simplewrapper
matplotlib.backend_bases.FigureCanvasBase
builtins.object
```
and `FigureCanvasAgg.__init__` is the same as `FigureCanvasBase.__init__`, which does *not* forward its arguments to the super class, so QWidget never sees the figure kwarg.

Or you can just test by yourself that it works :)